app/controllers/listings_controller.rb

```diff
@@ -51,16 +51,17 @@ def index
         page =  params[:page] || 1
         per_page = params[:per_page] || 50
 
+        all_shapes = get_shapes()
+        all_processes = get_processes()
```

Comment:
Are these two locals used outside the call to create direction_map? If not, I'd inline to help reader to see their only used for one purpose.